### PR TITLE
admin: Add node object to /server_info response

### LINKED ISSUE #9684: /server_info has empty service_[cluster|node|zone] when configured in bootstrap
This is a bug report that when one configures service_cluster, service_node, or service_zone using bootstrap configurations (on the `node` object) and not through CLI flags, then the admin interface reports these values as empty on`/server_info`. This is because the admin interface simply dumps whatever is contained in the `CommandLineOptions` struct (https://github.com/envoyproxy/envoy/blob/master/source/server/http/admin.cc#L929).

It would be best to fix this before https://github.com/envoyproxy/envoy/issues/3326

### PR DESCRIPTION
**Pull request by dio**
Commit Message: This patch adds node object information to GET /server_info response.
Risk Level: Low
Testing: Added
Docs Changes: Updated
Release Notes: Added
Fixes #9684

### CODE REVIEW

**Review by htuch — DISMISSED** [review]
LGTM, thanks!

File: include/envoy/local_info/local_info.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ class LocalInfo {
   virtual const std::string& nodeName() const PURE;
 
   /**
-   * v2 API Node protobuf. This is the full node identity presented to management servers.
+   * This is the full node identity presented to management servers.
```
Comment (jmarantz) [inline include/envoy/local_info/local_info.h:40]: `@return the full node identity presented to management servers.`

**Review by jmarantz — DISMISSED** [review]
@envoyproxy/senior-maintainers

File: include/envoy/local_info/local_info.h (hunk 1 above)
Comment (dio) [reply include/envoy/local_info/local_info.h]: Thanks for this. Updated.

**Comment by dio** [comment]
Merging main seems to fix the CI.

**Pull request merged**